### DISCUSSION TITLE
Abnormal structure image segmentation in rats

### DISCUSSION THREAD

**Discussion by Ernest861** [CoBrALab/RABIES#370] [Q&A]
I used BIDScoin to convert the T2 structure image it looks ok, 
<img width="1090" alt="Screenshot 2024-05-13 at 16 02 58" src="https://github.com/CoBrALab/RABIES/assets/28133972/ff0b09b6-43f5-4d86-aa29-b940f05bbe35">

then I found an error during processing, I don't know what the error is, can any friend help me to check it?
 
![sub-2_task-t2fsetra01x06mm_acq-t2fsetra01x06mm_echo-1_T2w_inho_cor](https://github.com/CoBrALab/RABIES/assets/28133972/e01ef2fc-edd6-4deb-8eb7-7d6f181516b7)

The T2 info:
“{
    "Modality": "MR",
    "MagneticFieldStrength": 9.4,
    "ImagingFrequency": 400.627,
    "Manufacturer": "UIH",
    "ManufacturersModelName": "uMR  9.4T",
    "DeviceSerialNumber": "H00011",
    "StationName": "UIHMR",
    "BodyPartExamined": "RATBRAIN",
    "PatientPosition": "HFS",
    "SoftwareVersions": "R002",
    "MRAcquisitionType": "2D",
    "SeriesDescription": "t2_fse_tra_0.1x0.6mm",
    "ProtocolName": "t2_fse_tra_0.1x0.6mm",
    "ScanningSequence": "SE",
    "SequenceVariant": "NONE",
    "SequenceName": "fse",
    "ImageType": [
        "ORIGINAL",
        "PRIMARY",
        "M",
        "IF"
    ],
    "SeriesNumber": 602,
    "AcquisitionTime": "14:32:12.286000",
    "SliceThickness": 0.6,
    "SpacingBetweenSlices": 0.96,
    "NumberOfAverages": 5,
    "EchoTime": 0.0462,
    "RepetitionTime": 3,
    "FlipAngle": 180,
    "PhaseEncodingDirectionDisplayed": "A->P",
    "EchoTrainLength": 13,
    "AcquisitionMatrixPE": 290,
    "ReconMatrixPE": 580,
    "TotalReadoutTime": 342.066,
    "PixelBandwidth": 220,
    "PhaseEncodingAxis": "j",
    "ImageOrientationPatientDICOM": [
        1,
        0,
        0,
        0,
        1,
        0
    ],
    "InPlanePhaseEncodingDirectionDICOM": "COL",
    "ConversionSoftware": "dcm2niix",
    "ConversionSoftwareVersion": "v1.0.20220505"
}”

**Comment by gdevenyi** [comment]
Please provide the exact commands you used to run the pipeline as well as all the logs.

**Reply by Ernest861** [reply]
Here is the commands: Whether using --anat_autobox or --anat_inho_cor method=SyN,otsu_thresh=2,multiotsu=true, the image is such that the structural states are enlarged and incorrectly delineated.
```240513-03:43:41,836 nipype.workflow INFO:
	 Running RABIES - version: 0.5.0
240513-03:43:41,837 nipype.workflow INFO:
	 CLI INPUTS: 
-> rabies_stage = preprocess 
-> plugin = MultiProc 
-> local_threads = 6 
-> scale_min_memory = 1.0 
-> min_proc = 1 
-> figure_format = png 
-> verbose = 1 
-> bids_dir = /test_dataset 
-> output_dir = /test_dataset_outputs 
-> bold_only = False 
-> anat_autobox = True 
-> bold_autobox = False 
-> apply_despiking = False 
-> HMC_option = optim 
-> isotropic_HMC = False 
-> voxelwise_motion = False 
-> apply_slice_mc = False 
-> detect_dummy = False 
-> data_type = float32 
-> anat_inho_cor = {'method': 'SyN', 'otsu_thresh': '2', 'multiotsu': False} 
-> anat_robust_inho_cor = {'apply': True, 'masking': True, 'brain_extraction': True, 'template_registration': 'SyN'} 
-> bold_inho_cor = {'method': 'Rigid', 'otsu_thresh': '2', 'multiotsu': False} 
-> bold_robust_inho_cor = {'apply': False, 'masking': False, 'brain_extraction': False, 'template_registration': 'SyN'} 
-> commonspace_reg = {'masking': False, 'brain_extraction': False, 'template_registration': 'SyN', 'fast_commonspace': False} 
-> bold2anat_coreg = {'masking': False, 'brain_extraction': False, 'registration': 'SyN'} 
-> nativespace_resampling = inputs_defined 
-> commonspace_resampling = inputs_defined 
-> anatomical_resampling = 0.15x0.15x0.15 
-> apply_STC = False 
-> TR = auto 
-> tpattern = alt-z 
-> stc_axis = Y 
-> interp_method = fourier 
-> anat_template = /home/rabies/.local/share/rabies/DSURQE_40micron_average.nii.gz 
-> brain_mask = /home/rabies/.local/share/rabies/DSURQE_40micron_mask.nii.gz 
-> WM_mask = /home/rabies/.local/share/rabies/DSURQE_40micron_eroded_WM_mask.nii.gz 
-> CSF_mask = /home/rabies/.local/share/rabies/DSURQE_40micron_eroded_CSF_mask.nii.gz 
-> vascular_mask = /home/rabies/.local/share/rabies/vascular_mask.nii.gz 
-> labels = /home/rabies/.local/share/rabies/DSURQE_40micron_labels.nii.gz 
```
attached log 
[rabies_preprocess.log](https://github.com/CoBrALab/RABIES/files/15317546/rabies_preprocess.log)

Thank you!

**Reply by gdevenyi** [reply]
`-> anat_template = /home/rabies/.local/share/rabies/DSURQE_40micron_average.nii.gz `

This is a mouse template, you need to select the SIGMA rat template.

**Comment by Ernest861** [comment]
Taking a hint from the forums [Using SIGMA instead of Dsurge](https://github.com/CoBrALab/RABIES/discussions/306) I used atlas from sigm and got the same error as [@KimMinhHuynh](https://github.com/CoBrALab/RABIES/discussions/369).

Traceback (most recent call last):
  File "/home/rabies/miniforge/bin/rabies", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/rabies/RABIES/scripts/rabies", line 3, in <module>
    execute_workflow()
  File "/home/rabies/RABIES/rabies/run_main.py", line 44, in execute_workflow
    workflow = preprocess(opts, log)
  File "/home/rabies/RABIES/rabies/run_main.py", line 202, in preprocess
    check_template_overlap(opts.anat_template, opts.CSF_mask)
  File "/home/rabies/RABIES/rabies/run_main.py", line 336, in check_template_overlap
    raise ValueError(
ValueError: The file /atlas_folder/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_CSF_mask.nii **does not appear to overlap with provided template** /atlas_folder/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_Brain_Template.nii.



the log file:
`c240516-09:19:31,688 nipype.workflow INFO:
	 Running RABIES - version: 0.5.0
240516-09:19:31,689 nipype.workflow INFO:
	 CLI INPUTS: 
-> rabies_stage = preprocess 
-> plugin = MultiProc 
-> local_threads = 6 
-> scale_min_memory = 1.0 
-> min_proc = 1 
-> figure_format = png 
-> verbose = 1 
-> bids_dir = /test_dataset 
-> output_dir = /test_dataset_outputs 
-> bold_only = False 
-> anat_autobox = True 
-> bold_autobox = False 
-> apply_despiking = False 
-> HMC_option = optim 
-> isotropic_HMC = False 
-> voxelwise_motion = False c
-> apply_slice_mc = False 
-> detect_dummy = False 
-> data_type = float32 
-> anat_inho_cor = {'method': 'SyN', 'otsu_thresh': '2', 'multiotsu': False} 
-> anat_robust_inho_cor = {'apply': True, 'masking': True, 'brain_extraction': True, 'template_registration': 'SyN'} 
-> bold_inho_cor = {'method': 'Rigid', 'otsu_thresh': '2', 'multiotsu': False} 
-> bold_robust_inho_cor = {'apply': False, 'masking': False, 'brain_extraction': False, 'template_registration': 'SyN'} 
-> commonspace_reg = {'masking': False, 'brain_extraction': False, 'template_registration': 'SyN', 'fast_commonspace': False} 
-> bold2anat_coreg = {'masking': False, 'brain_extraction': False, 'registration': 'SyN'} 
-> nativespace_resampling = inputs_defined 
-> commonspace_resampling = inputs_defined 
-> anatomical_resampling = 0.15x0.15x0.15 
-> apply_STC = False 
-> TR = auto 
-> tpattern = alt-z 
-> stc_axis = Y 
-> interp_method = fourier 
-> anat_template = /atlas_folder/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_Brain_Template.nii 
-> brain_mask = /atlas_folder/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_Brain_Mask.nii 
-> WM_mask = /atlas_folder/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_WM_mask.nii 
-> CSF_mask = /atlas_folder/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_CSF_mask.nii 
-> vascular_mask = /atlas_folder/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_Brain_Mask.nii 
-> labels = /atlas_folder/SIGMA_Rat_Brain_Atlases/SIGMA_Anatomical_Atlas/InVivo_Atlas/SIGMA_InVivo_Anatomical_Brain_Atlas.nii 

240516-09:19:31,689 nipype.workflow INFO:
	 INPUT BIDS DATASET:  
test_dataset/ 
    .bidsignore 
    sub-2/ 
        sub-2_scans.tsv 
        anat/ 
            sub-2_task-t2fsetra01x06mm_acq-t2fsetra01x06mm_echo-1_T2w.nii.gz 
            sub-2_task-t2fsetra01x06mm_acq-t2fsetra01x06mm_echo-1_T2w.json 
            testjoson/ 
        func/ 
            sub-2_task-epibold2shtra015x06mm_bold.nii.gz 
            sub-2_task-epibold2shtra015x06mm_bold.json 
    sub-9/ 
        sub-9_scans.tsv 
        anat/ 
            sub-9_task-t2fsetra01x06mm_acq-t2fsetra01x06mm_echo-1_T2w.nii.gz 
            sub-9_task-t2fsetra01x06mm_acq-t2fsetra01x06mm_echo-1_T2w.json 
        func/ 
            sub-9_task-epibold2shtra03x06mm_bold.nii.gz 
            sub-9_task-epibold2shtra03x06mm_bold.json 
`

**Reply by Gab-D-G** [reply]
Hi, this error will arise when the orientation or dimension of the images are not corresponding between the different atlas files. Can you verify voxel/matrix dimensions and orientations among your template files? the CSF mask is not corresponding to your template. Also take note that the masks must be binary images, and the SIGMA may be providing weighted CSF/WM maps.

**Reply by Ernest861** [reply]
I download the atlas file from https://www.nitrc.org/projects/sigma_template, all the voxel/matrix and ordinate in the invo version is same. 
attached files 
[altasfile.zip](https://github.com/CoBrALab/RABIES/files/15342482/altasfile.zip)

**Reply by gdevenyi** [reply]
The files are not all identical as is required:
```
$ for file in *nii; do echo ${file}; PrintHeader $file 1; done
SIGMA_InVivo_Anatomical_Brain_Atlas.nii
0.15x0.15x0.15
SIGMA_InVivo_Brain_Mask.nii
0.15x0.15x0.15
SIGMA_InVivo_Brain_Template.nii
0.15x0.15x0.15
SIGMA_InVivo_CSF_mask.nii
0.150002x0.149994x0.150002
SIGMA_InVivo_WM_mask.nii
0.15x0.15x0.15
```

**Reply by Ernest861** [reply]
Thanks for you information, I reslice the CSF mask and do the preprocess, but in the QC reports, it looks also have error.
I attached the log and files. Thank you for your time.
[preprocess_QC_report.zip](https://github.com/CoBrALab/RABIES/files/15368762/preprocess_QC_report.zip)
[rabies_preprocess.log](https://github.com/CoBrALab/RABIES/files/15368763/rabies_preprocess.log)

**Reply by Gab-D-G** [reply]
The orientation of your images doesn't seem to match that of the template. RABIES automatically resamples images to RAS orientation, but yours don't match, meaning the metadata is not well encoded. You may need to manually make sure your images are orientated in a way that is consistent with the template file.

**Reply by gdevenyi** [reply]
If you collecting your scans on a bruker, we recommend brkraw for conversion to ensure proper orientation (provided the orientation details were set properly in paravision)

**Reply by Ernest861** [reply]
Thank you both very much for the information, I collect the image by united-imaging uMR9.4T preclinical system.
I convert them from dicom to nii by [bidscoin](https://bidscoin.readthedocs.io/en/stable/). 
I gaze they use the cdm2niix2bids to convert, but i am not sure the parameter.
![bidscoin](https://github.com/CoBrALab/RABIES/assets/28133972/0598956d-51e8-45f9-91e0-4f47705b8f0a)

**Reply by gdevenyi** [reply]
dcm2niix excepts the DICOMs to be standards compliant regarding orientation of data. I am not aware of any options which can override orientation information. If something is wrong on NIFTI output, here are the possibilities.

1. The DICOMs have the wrong orientation info due to the DICOM export being wrong
2. The information regarding the animal was set wrong on the scanner (quadraped, Supine/Prone/Head-First/Feet First), but the DICOM export is correct

I suggest you do a few things
1. Check your scanner console settings are correct for the animal you scan
2. Check that the DICOM metadata regarding the type of sample (animal, positioning/orienation) is correct with something such as dcmdump
3. Open up the NIFTIs in itksnap to check their orientation. Correctly specified data will have the anatomical A-P, S-I and L-R labels in the display be correct with respect to the animal scan.

**Reply by Ernest861** [reply]
Thanks for the advice, I'll do all the steps on this one. 

The processing after manual reorientation is complete and it seems that the first short BOLD sequence is more correct and the second BOLD fails. I would like to confirm that the signal to noise ratio of the image is viewed temporal_features.png, is it possible to extract the values.

the first session
![sub-2_task-epibold2shtra015x06mm_bold_temporal_features](https://github.com/CoBrALab/RABIES/assets/28133972/710f9785-ae29-4d39-a610-8a3755d74221)

the second session
![sub-9_task-epibold2shtra03x06mm_bold_temporal_features](https://github.com/CoBrALab/RABIES/assets/28133972/b51a7924-d000-454a-9fa2-2ad046995b97)

attached file
[rabies_preprocess.log](https://github.com/CoBrALab/RABIES/files/15396702/rabies_preprocess.log)
[rabies_analysis.log](https://github.com/CoBrALab/RABIES/files/15396703/rabies_analysis.log)
[preprocess_QC_report.zip](https://github.com/CoBrALab/RABIES/files/15396706/preprocess_QC_report.zip)

**Reply by Gab-D-G** [reply]
This is correct, session 1 looks like registration worked, session 2 failed. You can use the tSNR map for session 1, although it looks like there are only 18 frames in your session 1, I'm not sure that's intended.
